.github/workflows/build-artifacts.yml

```diff
@@ -1,8 +1,12 @@
 name: build-artifacts.yml
+permissions:
+  id-token: write
+  contents: read
 on:
   workflow_dispatch:
   push:
-    branches: [ main, SERVER-216,  SERVER-434 ]
+    branches: [ SERVER-216 ]
```

Comment:
You can't test merges to main but shouldn't this run for PRs not just for testing but so PR commits get tested?